### PR TITLE
use managed entities in requestManager

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/RequestManager.java [hunk 1]
```diff
@@ -501,4 +497,10 @@
    */
   @Deprecated
   public Collection<LibraryDilution> listAllLibraryDilutions() throws IOException;
+
+  /**
+   * @Deprecated in v0.28. Use sampleService.save instead
+   */
+  @Deprecated
+  public long saveSample(Sample sample) throws IOException;
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/RequestManager.java]: Should just be removed, but that'll happen with switch-over to Services anyway

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java [hunk 2]
```diff
@@ -1130,18 +1148,28 @@ public long saveRun(Run run) throws IOException {
...
+        saveRun(run);
+      }
     } else {
       throw new IOException("No runStore available. Check that it has been declared in the Spring config.");
     }
   }
 
   @Override
-  public long saveRunQC(RunQC runQC) throws IOException {
+  public void saveRunQC(RunQC runQC) throws IOException {
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java]: The caller never consumed the output.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java [hunk 3]
```diff
@@ -1130,18 +1148,28 @@ public long saveRun(Run run) throws IOException {
...
     if (runQcStore != null) {
-      long id = runQcStore.save(runQC);
-      if (runAlertManager != null) runAlertManager.update(runQC.getRun());
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java]: Yes. Please revert.

Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java]: Reverted.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java (hunk 2 above)
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java]: It's easily accessible, can be useful, and all of the other save methods return the inserted ID. I don't see a reason to remove it, and would prefer consistency

**Pull request merged**